var/spack/repos/builtin/packages/rocfft/package.py

```diff
@@ -62,7 +62,7 @@ def cmake_args(self):
         # From version 3.9 and above we have AMDGPU_TARGETS_SRAM_ECC
         tgt_sram = self.spec.variants['amdgpu_target_sram_ecc'].value
 
-        if tgt_sram[0] != 'none' and '@3.9.0:' in self.spec:
+        if tgt_sram[0] != 'none' and self.spec.satisfies('@3.9.0:4.0.0'):
```

Comment:
This is unrelated to the PR? Can you open a separate PR for that?

Reply:
okay.. i have reverted the rocfft changes and will open a separate PR